crates/driver/src/infra/persistence/mod.rs

```diff
@@ -0,0 +1,70 @@
+use {
+    crate::{domain::competition::auction::Id, infra::solver::Config},
+    serde::Deserialize,
+    std::sync::Arc,
+    tracing::Instrument,
+};
+
+#[derive(Clone, Debug, Default, Deserialize)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields)]
+pub struct S3 {
+    /// The s3_instance_upload_* arguments configure how auction instances
+    /// should be uploaded to AWS S3.
+    /// They must either all be set or all not set.
+    pub bucket: String,
+
+    /// Prepended to the auction id to form the final instance filename on S3.
+    /// Something like "staging/mainnet/"
+    pub prefix: String,
+}
+
+impl From<S3> for s3::Config {
+    fn from(value: S3) -> Self {
+        Self {
+            bucket: value.bucket,
+            filename_prefix: value.prefix,
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Persistence {
+    s3: Option<Arc<s3::Uploader>>,
+}
+
+impl Persistence {
+    pub async fn build(config: &Config) -> Self {
+        if let Some(s3) = &config.s3 {
+            Self {
+                s3: Some(Arc::new(s3::Uploader::new(s3.clone().into()).await)),
+            }
+        } else {
+            Self { s3: None }
+        }
+    }
+
+    /// Saves the given auction with liquidity with fire and forget mentality
+    /// (non-blocking operation)
+    pub fn archive_auction(&self, auction_id: Option<Id>, body: &str) {
+        let Some(uploader) = self.s3.clone() else {
+            return;
+        };
+        let Some(id) = auction_id else {
+            return;
+        };
```

Comment:
Deciding that auctions without ids (i.e. quote requests) should not be uploaded seems like a domain decision to me. If `archive_auction` were to take a non-optional `Id` this would naturally be enforced by the type system in the domain logic.

Also there should be a comment somewhere explaining why we only store auctions which have an id.

Reply:
Sure! I'll move it to the domain logic. 
I believe (I may be wrong) that auctions without id are only the ones generated by `fake_auction` function :thinking:

Reply:
> I believe (I may be wrong) that auctions without id are only the ones generated by fake_auction function

Yes, that's right. Those are the quote requests which don't really make sense to store.